packages/zoe/test/unitTests/contracts/test-mintPayments.js

```diff
@@ -50,3 +52,72 @@ test('zoe - mint payments', async t => {
     console.log(e);
   }
 });
+
+test('zoe - mint payments with unrelated give and want', async t => {
+  t.plan(3);
+  try {
+    const zoe = makeZoe({ require });
+    // Pack the contract.
+    const { source, moduleFormat } = await bundleSource(mintPaymentsRoot);
+    const installationHandle = await E(zoe).install(source, moduleFormat);
+    const inviteIssuer = await E(zoe).getInviteIssuer();
+    const getInstanceHandle = makeGetInstanceHandle(inviteIssuer);
+
+    const moolaBundle = produceIssuer('moola');
+    const simoleanBundle = produceIssuer('simolean');
+
+    // Alice creates a contract instance
+    const issuerKeywordRecord = harden({
+      Asset: moolaBundle.issuer,
+      Price: simoleanBundle.issuer,
+    });
+    const adminInvite = await E(zoe).makeInstance(
+      installationHandle,
+      issuerKeywordRecord,
+    );
+    const instanceHandle = await getInstanceHandle(adminInvite);
+
+    // Bob wants to get 1000 tokens so he gets an invite and makes an
+    // offer
+    const { publicAPI } = await E(zoe).getInstanceRecord(instanceHandle);
+    const invite = await E(publicAPI).makeInvite();
+    t.ok(await E(inviteIssuer).isLive(invite), `valid invite`);
+    const proposal = harden({
+      give: { Asset: moolaBundle.amountMath.make(10) },
+      want: { Price: simoleanBundle.amountMath.make(100) },
```

Comment:
Bob's `want` was simoleans. Did he get any simoleans? 

I think this is being processed as a refund rather than a successful transaction where they got what they wanted. This shows that `reallocate()` worked. I'd also like to see that `complete()` works when there has been a trade.

Reply:
Yes, this is a refund because it is the makePayment contract, which doesn't do any swapping. Bob didn't want to get any simoleans.

Whether complete works when there is a trade is out of scope for this test, but it is tested by the atomicSwap test.